lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -883,34 +885,36 @@ def configure_network(self, vm_obj):
             return
 
         network_devices = list()
-        for network in self.params['networks']:
-            if network:
-                if 'ip' in self.params['networks'][network]:
-                    ip = self.params['networks'][network]['ip']
-                    if ip not in IPNetwork(network):
-                        self.module.fail_json(msg="ip '%s' not in network %s" % (ip, network))
-
-                    ipnet = IPNetwork(network)
-                    self.params['networks'][network]['subnet_mask'] = str(ipnet.netmask)
-
-                if 'network' in self.params['networks'][network]:
-                    if get_obj(self.content, [vim.Network], self.params['networks'][network]['network']) is None:
-                        self.module.fail_json(msg="Network %s doesn't exists" % network)
-                elif 'vlan' in self.params['networks'][network]:
-                    network_name = None
-                    dvps = get_all_objs(self.content, [vim.dvs.DistributedVirtualPortgroup])
-                    for dvp in dvps:
-                        if dvp.config.defaultPortConfig.vlan.vlanId == self.params['networks'][network]['vlan']:
-                            network_name = dvp.config.name
-                            break
-                    if network_name:
-                        self.params['networks'][network]['network'] = network_name
-                    else:
-                        self.module.fail_json(msg="VLAN %(vlan)s doesn't exists" % self.params['networks'][network])
+        for network in self.params.get('networks', list()):
+            if 'network' not in network:
```

Comment:
The old implementation gets the netmask from the network information. And that's how I was using it as well. Sometimes it is easier to provide the CIDR information instead of calculating the netmask.
(I don't mind, I know colleagues do though)

Let's see what we want to do with this. I am open to ideas.